Jobs.Fetcher.TikTok/Jobs/PostsQuery.cs

```diff
@@ -18,156 +20,266 @@ public class PostsQuery : AbstractTikTokFetcher {
             return new List<string>() {};
         }
 
-        public override void RunBody(string username) {
+        private readonly object MyLock = new object();
+
+        private int RetrievedInFull = 0;
+        private int MissingSuperfluous = 0;
+        private int MissingEssential = 0;
+
+        private void FoundData(string fieldName, string postId) {
+            Logger.Information($"🟢 id = {postId} - '{fieldName}' retrieved in full!");
+            lock (MyLock) RetrievedInFull++;
+        }
+
+        private void MissingSuperfluousData(string fieldName, string postId) {
+            Logger.Warning($"🟡 id = {postId} - could not retrieve '{fieldName}', it won't be saved");
+            lock (MyLock) MissingSuperfluous++;
+        }
```

Comment:
This function can be called a few different times while getting the same Post, right? So assuming we are fetching only 2 posts and it is missing 3 superfluous parts I would get as a result:
2 posts processed
1 complete post
3 missing superfluous
Or am I missing something?